core/lib/dal/src/connection/mod.rs

```diff
@@ -150,17 +150,17 @@ impl ConnectionPool {
     }
 
     /// Initializes a builder for connection pools.
-    pub fn builder(database_url: &str, max_pool_size: u32) -> ConnectionPoolBuilder<'_> {
+    pub fn builder(database_url: &str, max_pool_size: u32) -> ConnectionPoolBuilder {
         ConnectionPoolBuilder {
-            database_url,
+            database_url: database_url.to_string(),
```

Comment:
Micro-nit: I think `to_owned()` is more canonical way to express this conversion (there's [a Clippy lint](https://rust-lang.github.io/rust-clippy/master/index.html#/str_to_string) for this).

Reply:
I personally find this lint controversial, unless enforced throughout the workspace.
If/when we will get to the point where we enable it, I'll be more than happy to follow Clippy's suggestions (e.g. I'm sure that *everyone* follows it and the codebase is consistent), but without it, I think it just adds to the potential nitpicking reason pool (which isn't small anyways), making reviews longer without any particular benefit.